cloud/services/compute/firewalls.go

```diff
@@ -67,6 +67,25 @@ func (s *Service) DeleteFirewalls() error {
 		delete(s.scope.Network().FirewallRules, name)
 	}
 
+	//delete any additional non-default firewall rules that were generated
+	firewallRules, err := s.firewalls.List(s.scope.Project()).Do()
+	if err != nil {
+		return errors.Wrapf(err, "failed to list firewall rules for project %s", s.scope.Project())
+	}
+
+	clusterName := s.scope.Name()
+	for _, firewall := range firewallRules.Items {
+		for _, tt := range firewall.TargetTags {
+			//only delete rules with target-tags containing the cluster-name
```

Comment:
shall we also limit the rules to the network?

Reply:
ack